internal/sql/repository/AppListingRepository.go

```diff
@@ -496,6 +500,17 @@ func (impl AppListingRepositoryImpl) FindLastDeployedStatusesForAllApps() ([]Dep
 	return deploymentStatuses, nil
 }
 
+func (impl AppListingRepositoryImpl) FindLastDeployedStatusesForAllActiveAppsWithActiveEnv() ([]DeploymentStatus, error) {
+	var deploymentStatuses []DeploymentStatus
+	query := "select distinct ds.app_name, status, max(ds.id) as id, app_id, env_id, ds.created_on, ds.updated_on from deployment_status ds inner join app on app.id=ds.app_id inner join environment env on env.id=ds.env_id where app.active=true and env.active=true group by ds.app_name, status,app_id, env_id, ds.created_on, ds.updated_on order by id desc;"
```

Comment:
filter-out data based on deployment time and status to reduce list size otherwise for large users it will create performance issue